pkg/auth/providers/azure/clients/ms_graph_client.go

```diff
@@ -282,6 +293,15 @@ func NewMSGraphClient(config *v32.AzureADConfig, secrets corev1.SecretInterface)
 			return nil, err
 		}
 	}
+
+	if err = EnsureMSGraphTokenHasPermissions(ar.AccessToken); err != nil {
+		if deletionErr := tokenCache.Clear(); deletionErr != nil {
+			logrus.Errorf("[%s] failed to delete the cached secret %s with an invalid access token: %v",
```

Comment:
Typically we don't like to signal the location of cache secrets @cbron any opinion on this error message?

Reply:
I'm ok with it being in logs, since those should be admin-only anyway. The code is public so if someone really wants to know they can figure it out. As long as we aren't logging any actual data, which it looks like we are not.